sechub-wrapper-owasp-zap/src/main/java/com/mercedesbenz/sechub/zapwrapper/helper/ZapProductMessageHelper.java

```diff
@@ -58,14 +60,23 @@ public void writeProductError(ZapWrapperRuntimeException zapWrapperRuntimeExcept
         }
     }
 
-    public void writeUserMessagesWithScannedURLs(List<String> urls) {
-        for (String url : urls) {
+    public void writeUserMessagesWithDetectedURLs(List<Map<String, ApiResponse>> results) {
+        // The map looks like this: {processed=true, statusReason=OK, method=GET,
+        // reasonNotProcessed=, messageId=6, url=http://example.com, statusCode=200}
+        for (Map<String, ApiResponse> result : results) {
+            String url = result.get("url").toString();
             // robots.txt and sitemap.xml always appear inside the sites tree even if they
             // are not available. Because of this it is skipped here.
             if (url.contains("robots.txt") || url.contains("sitemap.xml")) {
                 continue;
             }
-            writeSingleProductMessage(new SecHubMessage(SecHubMessageType.INFO, "Detect url to scan: " + url));
+            String statusCode = result.get("statusCode").toString();
+            String statusReason = result.get("statusReason").toString();
+            String method = result.get("method").toString();
+
+            String message = "URL: '%s' returned status code: '%s/%s' on detection phase for request method: '%s'".formatted(url, statusCode, statusReason,
+                    method);
+            writeSingleProductMessage(new SecHubMessage(SecHubMessageType.INFO, message));
```

Comment:
IMO this can handle to an extreme amount of messages!
Let's assume ZAP has scanned 2000 URLs -> we have 2000 entries here.
Even when the scans were for the same URL but with different content!

Should we filter/reduce here to avoid duplicates?